app/src/main/java/io/capibaras/abcall/viewmodels/LoginViewModel.kt

```
@@ -66,40 +64,35 @@ class LoginViewModel(
-                        ErrorUIState.Error(R.string.error_incorrect_credentials)
+                    if (response.code() == 401) {
+                        stateMediator.setErrorState(ErrorUIState.Error(R.string.error_incorrect_credentials))
                     } else {
                         val errorBody = response.errorBody()!!.string()
                         val jsonObject = JSONObject(errorBody)
                         val message = jsonObject.getString("message")
-                        ErrorUIState.Error(message = message)
+                        stateMediator.setErrorState(ErrorUIState.Error(message = message))
                     }
                 }
             } catch (e: IOException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)